rmf_human_detector/src/YoloDetector.cpp

```diff
@@ -0,0 +1,439 @@
+#include <memory>
+#include <cstdio>
+#include <fstream>
+#include <cmath>
+#include <filesystem>
+#include <memory>
+
+// ROS includes
+#include <std_msgs/msg/header.hpp>
+#include <builtin_interfaces/msg/time.hpp>
+#include <builtin_interfaces/msg/duration.hpp>
+#include <geometry_msgs/msg/point.hpp>
+#include <geometry_msgs/msg/quaternion.hpp>
+#include <geometry_msgs/msg/pose.hpp>
+#include <geometry_msgs/msg/vector3.hpp>
+#include <geometry_msgs/msg/pose_stamped.hpp>
+#include <geometry_msgs/msg/vector3_stamped.hpp>
+#include <geometry_msgs/msg/transform_stamped.hpp>
+#include <tf2/LinearMath/Quaternion.h>
+#include <tf2/LinearMath/Matrix3x3.h>
+#include <tf2_ros/buffer.h>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+
+// OpenCV includes
+#include <cv_bridge/cv_bridge.h>
+
+// Project includes
+#include <rmf_obstacle_msgs/msg/bounding_box3_d.hpp>
+#include <rmf_human_detector/YoloDetector.hpp>
+
+// Namespaces
+using namespace cv;
+using namespace std;
+using namespace cv::dnn;
+
+// Text parameters
+const float FONT_SCALE = 0.4;
+const int FONT_FACE = FONT_HERSHEY_SIMPLEX;
+const int FONT_THICKNESS = 1;
+const int RECT_THICKNESS = 2;
+
+// Colors
+const Scalar BLACK = Scalar(0,0,0);
+const Scalar BLUE = Scalar(255, 178, 50);
+const Scalar YELLOW = Scalar(0, 255, 255);
+const Scalar RED = Scalar(0,0,255);
+
+YoloDetector::YoloDetector(std::shared_ptr<Config> config) : _config(config)
+{
+    calibrate();
+    if (_config->visualize)
```

Comment:
This _config->visualize variable is used to control if the user wants to visualize the final image with bounding boxes as an OpenCV image. Either way, the final image is being published as a ros topic.